src/states/approved_peer/mod.rs

```diff
@@ -51,6 +52,9 @@ use std::{
     net::SocketAddr,
 };
 
+// Send our knowledge in a similar speed as GOSSIP_TIMEOUT
+const KNOWLEDGE_TIMEOUT: Duration = Duration::from_secs(2);
```

Comment:
Possibly, but this is how it is in fleming and I'd prefer not to change it in this PR. Also the plan is to eventually not use any timeout and make the `MemberKnowledge` event-based instead of time-based. But there are some corner cases that neat to be handled first. In any case, it's not in the scope of this PR anyway.